src/Fields/field_tuples.jl

```diff
@@ -57,11 +57,46 @@ function fill_halo_regions!(maybe_nested_tuple::Union{NamedTuple, Tuple}, args..
     flattened = flattened_unique_values(maybe_nested_tuple)
 
     # Look for grid within the flattened field tuple:
-    grid = flattened[1].grid
+    for f in flattened
+        if isdefined(f, :grid)
+            grid = f.grid
+            return tupled_fill_halo_regions!(flattened, grid, args...; kwargs...)
+        end
+    end
+
+    return tupled_fill_halo_regions!(flattened, args...; kwargs...)
+end
 
-    return tupled_fill_halo_regions!(flattened, grid, args...; kwargs...)
+# Version where we find grid amongst ordinary fields:
+function tupled_fill_halo_regions!(fields, args...; kwargs...)
+
+    ordinary_fields = Field[]
+    for f in fields
+        if !isnothing(boundary_conditions(f))
+            if f isa ReducedField || !(f isa FullField)
+                # Windowed and reduced fields
+                fill_halo_regions!(f, args...; kwargs...)
+            else
+                push!(ordinary_fields, f)
+            end
+        end
+    end
+
+    ordinary_fields = tuple(ordinary_fields...)
+
+    if !isempty(ordinary_fields) # ie not reduced, and with default_indices
+        grid = first(ordinary_fields).grid
+        fill_halo_regions!(map(data, ordinary_fields),
+                           map(boundary_conditions, ordinary_fields),
+                           default_indices(3),
+                           map(instantiated_location, ordinary_fields),
+                           grid, args...; kwargs...)
+    end
+
+    return nothing
 end
     
+# Version where grid is provided:
 function tupled_fill_halo_regions!(fields, grid, args...; kwargs...)
```

Comment:
```suggestion
function tupled_fill_halo_regions!(fields, grid::AbstractGrid, args...; kwargs...)
```

Reply:
otherwise this is not a different type signature from the function above if there are any `args...`